components/crud-web-apps/common/frontend/kubeflow-common-lib/projects/kubeflow/src/lib/resource-table/table/utils.ts

```diff
@@ -58,4 +58,99 @@ export function formatBytes(bytes, si = false, decimals = 1) {
   );
 
   return bytes.toFixed(decimals) + ' ' + units[u];
-}
\ No newline at end of file
+}
+
+export function findSuffix(quantity: string): string {
+  let ix = quantity.length - 1;
+  while (ix >= 0 && !/[\.0-9]/.test(quantity.charAt(ix))) {
+    ix--;
+  }
+  return ix === -1 ? '' : quantity.substring(ix + 1);
+}
+
+export function quantityToScalar(quantity: string): number | bigint {
+  if (!quantity) {
+    return 0;
+  }
+  const suffix = findSuffix(quantity);
+  if (suffix === '') {
+    const num = Number(quantity).valueOf();
+    if (isNaN(num)) {
+      throw new Error('Unknown quantity ' + quantity);
+    }
+    return num;
+  }
+  switch (suffix) {
+    case 'n':
+      return (
+        Number(quantity.substr(0, quantity.length - 1)).valueOf() /
+        1_000_000_000.0
+      );
+    case 'u':
+      return (
+        Number(quantity.substr(0, quantity.length - 1)).valueOf() / 1_000_000.0
+      );
+    case 'm':
+      return Number(quantity.substr(0, quantity.length - 1)).valueOf() / 1000.0;
+    case 'k':
+      return Number(quantity.substr(0, quantity.length - 1)) * Number(1000);
+    case 'M':
+      return (
+        Number(quantity.substr(0, quantity.length - 1)) * Number(1000 * 1000)
+      );
+    case 'G':
+      return (
+        Number(quantity.substr(0, quantity.length - 1)) *
+        Number(1000 * 1000 * 1000)
```

Comment:
@elenzio9 Nit but is there a reason for using `*` operator multiple times instead of the power one `**`? If not, I think `**` is more intuitive for this case (e.g. `1000 ** 3`)

Reply:
The `quantityToScalar()` is based on this [function](https://github.com/kubernetes-client/javascript/blob/14b6d48151f838df6c395472609e486ebed6c100/src/util.ts#L17) of `kubernetes-client` library. We only needed to replace `function`'s `BigInt()` with just `Number()`, since the `BigInt()` doesn't accept floats and in our case users can insert such numbers. 

However, it doesn't cause precision problems, as the [maximum safe integer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER) which can be represented by the `number` primitive is `2^53 – 1` and we are not interested in numbers larger than that. 

@orfeas-k For this reason, I'd suggest keeping it as is.

Reply:
Ok I see, that's fine by me then. Thanks for the context @elenzio9